doc/test_messages_documentation.py

```diff
@@ -91,7 +91,7 @@ def __init__(self, test_file: Tuple[str, Path]) -> None:
             args_list=[
                 str(test_file[1]),
                 "--disable=all",
-                f"--enable={test_file[0]}",
+                f"--enable={test_file[0]},astroid-error,fatal,syntax-error",
```

Comment:
Ah, I don't think we want to enable all errors, though. Just the fatal ones from pylint/astroid breaking, not antipatterns in the code under test.

Reply:
What would be an example of error we want to keep when we're not specifically documenting it ?

Reply:
Good question. I thought I remembered the original discussion was "don't try to pass every pylint check" but you have a point that maybe errors are worth avoiding. Things like `undefined-variable` can be tedious to avoid if you have initialize all your `thing` and `my_list` etc. Let me see what fails if we add this.

Reply:
some examples (there are more):
```
doc/data/messages/w/wrong-import-order/good.py:6:0: E0611: No name 'utils' in module '' (no-name-in-module)
doc/data/messages/y/yield-inside-async-function/good.py:6:0: E0102: function already defined line 1 (function-redefined)
```

Reply:
Hmm yeah wrong-import-order is possible to do if we remove the "our own lib" part but then it would be incomplete. For the other one we might need to find a way to propose multiple solution to the same bad.py but this is a non trivial refactor to do. I was thinking generally that being forced to create example without error would make the examples better but let's revisit that later when the cost to do it is lower.

Reply:
The boundary is probably subjective. I asked a contributor to fix an easy TypeError from not passing an argument to a function in #6472

Reply:
We could also explicitly disable the errors in the example ? (``# pylint: disable=no-name-in-module`` for example) Would prevent us from not seeing that there's an issue in an example.